R-package/R/multithreading.R

```diff
@@ -0,0 +1,51 @@
+#' @name setLGBMThreads
+#' @title Set maximum number of threads used by LightGBM
+#' @description LightGBM attempts to speed up many operations by using multi-threading.
+#'              The number of threads used in those operations can be controlled via the
+#'              \code{num_threads} parameter passed through \code{params} to functions like
+#'              \link{lgb.train} and \link{lgb.Dataset}. However, some operations (like materializing
+#'              a model from a text file) are done via code paths that don't explicitly accept thread-control
+#'              configuration.
+#'
+#'              Use this function to set the maximum number of threads LightGBM will use for such operations.
+#'
+#'              This function affects all LightGBM operations in the same process.
+#'
+#'              So, for example, if you call \code{setLGBMthreads(4)}, no other multi-threaded LightGBM
+#'              operation in the same process will use more than 4 threads.
+#'
+#'              Call \code{setLGBMthreads(-1)} to remove this limitation.
+#' @param num_threads maximum number of threads to be used by LightGBM in multi-threaded operations
+#' @return NULL
+#' @seealso \link{getLGBMthreads}
+#' @export
+setLGBMthreads <- function(num_threads) {
+    .Call(
+        LGBM_SetMaxThreads_R,
+        num_threads
+    )
+    return(invisible(NULL))
+}
```

Comment:
These functions are currently just setters and getters for `LGBM_MAX_NUM_THREADS` threads.

1. should they be named `getMaxLGBMthreads()` / `setMaxLGBMthreads()` or something else with "max" in the name?
    - *I personally like `getLGBMthreads()` / `setLGBMthreads()` for consistency with `data.table::{get/set}DTthreads()`, but could be convinced*
2. Should `getLGBMthreads()` even be exported in the R package's public API?
    - *I found it useful for testing and thought users might as well, but it'd be easier to add it later than to have to remove it later*
    - *and `{lightgbm}` could use it in its own tests with `:::`*
3. if we do keep it in the public interface... should `getLGBMthreads()` actually be a getter for `LGBM_MAX_NUM_THREADS`? Or should it return an answer to the question *"how many threads will e.g. `lgb.train()` use if I don't pass any thread-control parameters through `params`"*?